### PR TITLE
feat: Split paper outline into section files

### PR DESCRIPTION
**Pull request by matthewfeickert**
To make it easier for the authors to write together on the document without creating merge conflicts, split the document out into multiple section files that get compiled together.

The goal it to be able to include sections in a similar way as you would with the `\include` directive in LaTeX.

```
* Use the mystmd include directive to insert the contents of multiple source
  files into the document.
   - c.f. https://mystmd.org/guide/directives#directive-include
* Exclude the source files from the project in myst.yml to avoid them
  being rendered as additional supporting documents.
```

### CODE REVIEW

File: paper/main.md [hunk 1]
```diff
@@ -55,7 +55,9 @@ Furthermore, library developers will have the chance to see how their products a
 
 ## Introduction
 
-_Here we can talk about the ATLAS experiment, the detector, its purpose, the amount of data collected, etc._
+How can the contents of `paper/introduction.md` get inserted here in a similar manner to an include directive in LaTeX?
+
+It is unclear how to get myst to build multiple files into a single view like shown in jupyterbook: https://jupyterbook.org/en/stable/structure/toc.html#build-an-article-from-multiple-files as there is no `--builder` option for `myst start` or `myst build`.
```
Comment (matthewfeickert) [inline paper/main.md]: @rowanc1 while I'll ask elsewhere as well (probably Discord or Discussions), as this is SciPy 2024 related, can you confirm that what I want to do is possible and if so, point to docs on how?

What I currently have in this PR still creates separate pages of supporting documents

image

image

Comment (matthewfeickert) [reply paper/main.md]: Asked as https://github.com/orgs/executablebooks/discussions/1182

Comment (matthewfeickert) [reply paper/main.md]: Resolved by @agoose77 in https://github.com/orgs/executablebooks/discussions/1182#discussioncomment-9540467!

Comment (rowanc1) [reply paper/main.md]: Hi @matthewfeickert -- just got this notification now (was on vacation), glad that you are through this!

**Pull request merged**